generators/client/templates/react/src/main/webapp/app/modules/account/password/password-strength-bar.tsx

```diff
@@ -0,0 +1,79 @@
+import './password-strength-bar.<%= styleSheetExt %>';
+
+import * as React from 'react';
+import { Translate } from 'react-jhipster';
+
+export interface IPasswordStrengthBarProps {
+  password: string;
+}
+
+export default class PasswordStrengthBar extends React.Component<IPasswordStrengthBarProps> {
+  colors = ['#F00', '#F90', '#FF0', '#9F0', '#0F0'];
+
+  measureStrength(p: string): number {
+    let force = 0;
+    const regex = /[$-/:-?{-~!"^_`\[\]]/g; // "
+    const lowerLetters = /[a-z]+/.test(p);
+    const upperLetters = /[A-Z]+/.test(p);
+    const numbers = /[0-9]+/.test(p);
+    const symbols = regex.test(p);
+
+    const flags = [lowerLetters, upperLetters, numbers, symbols];
+    const passedMatches = flags.filter((isMatchedFlag: boolean) => isMatchedFlag === true).length;
```

Comment:
this might be cleaner
```
const flags = {
    lowerLetters = /[a-z]+/.test(p),
    upperLetters = /[A-Z]+/.test(p),
    numbers = /[0-9]+/.test(p),
    symbols = regex.test(p)
};

const passedMatches = Object.values(flags).filter((isMatchedFlag: boolean) => !!isMatchedFlag).length;
```